MAINTAINING.md

```diff
@@ -0,0 +1,147 @@
+# RIOT Review Guidelines
+
+The items presented in the following list should be checked for each PR review,
+in this order. This list is however not exhaustive and defines a rather minimal
+baseline on things that should be checked.
+
+Note: Any of the items in this list might be skipped, if **good** reasons are presented.
+
+Also note: This reviewing order is meant to maximize efficiency and minimize overhead, since failing in step n makes steps n+x obsolete. Maintainers should be aware that starting a review with 4. (Code Style) not only wasts their own time, but also the time of the author
+
+## 1. - Review the fundaments
+
+1.1 Does the reasoning for this PR make sense?
+1.2 Do the concepts used by the PR make sense? 
+1.3 Does the PR break with existing concepts?
+1.4 Is the structure of the PR itself valid?
+1.5 Are the testing criteria clear?
+(1.6 Does this PR respect the rights of previous authors?)
+(1.7 Is the PR a duplicate of another PR?)
+
+
+## 2. - Review the code (line-by-line)
+
+The following list is not exhaustive and addresses the coding issues we have
+regularly seen in the past.
+
+2.1 Check for code duplication
+2.2 Check memory usage
+2.3 Check all code paths
+2.4 Check for API compliance
+2.5 Check for consistent error handling
+2.6 Check scope of variables and functions
+2.7 Check for programming errors
+2.8 Check for code efficiency
```

Comment:
Though I don't know which efficiency is meant here, that point seems a bit generic to me. Above was already mentioned the code duplication or memory efficiency  so this refers to run efficient programming or run time efficiency? Do we want to mention that?

Reply:
Yep that sounds like it, could replace "code efficiency" with "run time efficiency".

Reply:
Hmm yeah. I guess with this point, it's a balance. Could be phrased, "check that code doesn't have any obvious, unnecessary inefficiencies"? I.e. we're checking here for things that are plain wrong, eg copying a whole struct to a function instead of handing a pointer?

Reply:
We should com up with and point to a document defining some best-practices.

There are things that we'd do differently than if we'd not be in tight embedded space.
E.g., if docs specify that argument three needs to point to a buffer that can hold N bytes, we only "assert()" whether the pointer is NULL instead of returning an error at runtime for production builds. That goes against safety / reliability best practices for any serious C development on "real" computers, but is necessary for us in order to keep code size down.

Maybe add something like "(See best practices here: TBD)" would make sense?

Reply:
> "(See best practices here: TBD)"

I've just seen the best practices wiki entry that @PeterKietzmann linked. Down the line, that should have an "efficiency" part that we could link to.

Reply:
+1 for developing the "best practices". This could be a useful focal point to develop our shared understanding of design priorities